### DISCUSSION TITLE
bug

### DISCUSSION THREAD

**Discussion by ChengJunJunJun** [dynamicslab/pysindy#340] [Q&A] [answered]
it remind me that module 'pysindy' has no attribute 'SINDyPI',but I just followed your example
<img width="853" alt="截屏2023-05-21 21 59 40" src="https://github.com/dynamicslab/pysindy/assets/127819698/dc89f562-a3c2-48ff-bcc5-80c34a3da5a5">

**Comment by Jacob-Stevens-Haas** [comment, marked as answer]
What error occurs if you run

```python
from pysindy.optimizers.sindy_pi import SINDyPI
```

**Reply by ChengJunJunJun** [reply]
![image](https://github.com/dynamicslab/pysindy/assets/127819698/75d71e89-9e3d-443f-91a7-2aa34b2375d4)
I will try to  install cvxpy

**Reply by ChengJunJunJun** [reply]
thank you somuch

**Reply by Jacob-Stevens-Haas** [reply]
You're very welcome.  There's a few difficult dependencies that we try to manage as optional dependencies.  We can probably guard them better with more informative errors and warnings.
